src/tests/suite/dfs_par_test.c

```diff
@@ -880,7 +880,14 @@ dfs_setup(void **state)
 	if (arg->myrank == 0) {
 		dfs_attr_t attr = {};
 
+		attr.da_props = daos_prop_alloc(1);
+		assert_non_null(attr.da_props);
+		attr.da_props->dpp_entries[0].dpe_type =
+					DAOS_PROP_CO_EC_CELL_SZ;
+		attr.da_props->dpp_entries[0].dpe_val = 1 << 15;
+
```

Comment:
Hmm, I think a few original test assume the cell size is 1M here, is it ok to change it to be 32K here.  

Probably confirm with the original writer?

Reply:
i run dfs_test locally it looks pass:
=================
DFS parallel tests..
=====================
[==========] Running 6 test(s).
setup: creating pool, SCM size=4 GB, NVMe size=16 GB
setup: created pool 2984f887-f7e9-4b79-aa11-6b40d42ebbe5
setup: connecting to pool 2984f887-f7e9-4b79-aa11-6b40d42ebbe5
connected to pool, ntarget=24
Created DFS Container c5a235cb-5b1b-411c-a655-6d8a2a95aa5c
[ RUN      ] DFS_PAR_TEST1: Conditional OPs
Testing with RF 0 ...
Created DFS Container c2382feb-dfea-4d3c-b2ab-56902cd1d641
All ranks create the same file with O_EXCL
All ranks unlink the same file
All ranks create the same directory
All ranks remove the same directory
Destroyed DFS Container c2382feb-dfea-4d3c-b2ab-56902cd1d641
Testing with RF 1 ...
Created DFS Container f7944084-c2c3-465e-84d9-0a30d0a8654c
All ranks create the same file with O_EXCL
All ranks unlink the same file
All ranks create the same directory
All ranks remove the same directory
Destroyed DFS Container f7944084-c2c3-465e-84d9-0a30d0a8654c
Testing with RF 2 ...
Created DFS Container bacab8fc-93df-4acd-ba99-37b784605c1b
All ranks create the same file with O_EXCL
All ranks unlink the same file
All ranks create the same directory
All ranks remove the same directory
Destroyed DFS Container bacab8fc-93df-4acd-ba99-37b784605c1b
[       OK ] DFS_PAR_TEST1: Conditional OPs
[ RUN      ] DFS_PAR_TEST2: DFS short reads
[       OK ] DFS_PAR_TEST2: DFS short reads
[ RUN      ] DFS_PAR_TEST3: DFS EC object short reads
[       OK ] DFS_PAR_TEST3: DFS EC object short reads
[ RUN      ] DFS_PAR_TEST4: DFS hole management
[       OK ] DFS_PAR_TEST4: DFS hole management
[ RUN      ] DFS_PAR_TEST5: DFS Container create atomicity
All ranks create the same POSIX container
one rank Created POSIX Container dfs_par_test_cont
Destroyed POSIX Container dfs_par_test_cont
[       OK ] DFS_PAR_TEST5: DFS Container create atomicity
[ RUN      ] DFS_PAR_TEST6: DFS File create (without O_EXCL) atomicity
All ranks create the same file without O_EXCL
Testing with RF 0 ...
Created DFS Container 6652e6ed-5ed0-44da-bfd7-27008563b84d
remove the file
reopen the file with OCREAT from all ranks
remove the file
Destroyed DFS Container 6652e6ed-5ed0-44da-bfd7-27008563b84d
Testing with RF 1 ...
Created DFS Container bb59a328-cc33-446b-a54a-61afea24b4a4
remove the file
reopen the file with OCREAT from all ranks
remove the file
Destroyed DFS Container bb59a328-cc33-446b-a54a-61afea24b4a4
Testing with RF 2 ...
Created DFS Container 5e5a5e30-a09f-416e-ad85-277e44c3e1d2
remove the file
reopen the file with OCREAT from all ranks
remove the file
Destroyed DFS Container 5e5a5e30-a09f-416e-ad85-277e44c3e1d2
[       OK ] DFS_PAR_TEST6: DFS File create (without O_EXCL) atomicity
Destroyed DFS Container c5a235cb-5b1b-411c-a655-6d8a2a95aa5c
teardown: destroyed pool 2984f887-f7e9-4b79-aa11-6b40d42ebbe5
[==========] 6 test(s) run.
[  PASSED  ] 6 test(s).

But looking at testing codes, something like:
dfs_test_ec_short_read:
        /* less than 1 EC stripe */
        dfs_test_short_read_internal(state, OC_EC_4P2G1,
                                     32 * 1024 * 8, 2000);

        /* partial EC stripe */
        dfs_test_short_read_internal(state, OC_EC_4P2G1,
                                     32 * 1024 * 8, 32 * 1024 * 2);

        /* full EC stripe */
        dfs_test_short_read_internal(state, OC_EC_4P2G1,
                                     32 * 1024 * 8, 32 * 1024 * 4);

        /* one full EC stripe + partial EC stripe */
        dfs_test_short_read_internal(state, OC_EC_4P2G1,
                                     32 * 1024 * 8, 32 * 1024 * 6);

        /* 2 full stripe */
        dfs_test_short_read_internal(state, OC_EC_4P2G1,
                                     32 * 1024 * 8, 32 * 1024 * 6);
I guess there might be some assuming that CELL size is 1M? I am not sure what it means chunk size 32 * 1024 *8 vs 1M cell size here @mchaarawi any idea for this?

Reply:
i was not the one who write the ec tests in the DFS suite. maybe @wangdi1  or @liuxuezhao?
that being said, chunk size is not related to the cell size. the array chunk size for the file, is how much bytes are writing underneath to a single dkey before moving to the next dkey.

Reply:
the original code in dfs_par_test.c use _L32K EC obj class, so seems fine.